src/agent/srcview/src/pdbcache.rs

```diff
@@ -47,57 +133,74 @@ impl PdbCache {
-                        let file_name = file_info.name.to_string_lossy(&string_table)?;
-
-                        let path = file_name.into_owned();
-                        let line = line_info.line_start as usize;
-
-                        let srcloc = SrcLine::new(path.clone(), line);
-
-                        offset_to_line.insert(rva.0 as usize, srcloc.clone());
-                        path_to_symbols
-                            .entry(path.clone())
-                            .or_default()
-                            .push(proc_name.to_string());
-                        symbol_to_lines.push(srcloc.clone());
-                        path_to_lines.entry(path.clone()).or_default().push(line);
+                match symbol.parse() {
+                    Ok(SymbolData::Procedure(proc)) => {
+                        let lines = program.lines_for_symbol(proc.offset);
+                        let proc_name = proc.name.to_string();
+                        builder.update_from_iter(&program, &string_table, &address_map, &proc_name, lines)?;
+                    }
+                    Ok(SymbolData::InlineSite(site)) => {
+                        // Locate the parent proc of this inline site
+                        let mut sid = site.parent;
+                        let offset;
+                        loop {
+                            let sid_actual = sid.ok_or_else(|| format_err!("S_INLINESITE should have parent"))?;
+                            let mut tmp_iter = info.symbols_at(sid_actual)?;
+                            let data = tmp_iter.next()?
+                                .ok_or_else(|| format_err!("Unable to parse symbol at {sid_actual}"))?
+                                .parse()?;
+                            match data {
+                                SymbolData::Procedure(proc) => {
+                                    offset = proc.offset;
+                                    break;
+                                }
+                                SymbolData::InlineSite(isite) => {
+                                    sid = isite.parent;
+                                }
+                                _ => bail!("Expected S_INLINESITE or procedure symbol"),
+                            }
+                        }
+
+                        let (inlinee, fname) = inlinees.get(&site.inlinee)
+                            .ok_or_else(|| format_err!("Cannot find inlinee for {}", site.inlinee))?;
+                        // NOTE: BA_OP_ChangeCodeOffsetBase is actually wrong in pdb crate package
+                        // See https://dev.azure.com/mseng/LLVM/_versionControl?path=%24/LLVM/pu/WinC/vctools/PDB/dia2/symcache.cpp&version=T&line=3453&lineEnd=3453&lineStartColumn=18&lineEndColumn=36&lineStyle=plain&_a=contents
+                        // for what it should actually be. The value is actually an index into the
+                        // nth S_SEPCODE entry which contains a section + offset value
```

Comment:
Are we contributing a fix for this as well?

Reply:
Not sure tbh, since I'm not even sure that msvc actually emits `BA_OP_ChangeCodeOffsetBase` opcodes anyways in the PDB